qgreenland/models/base_model.py

```diff
@@ -20,3 +21,8 @@ class Config:
         # https://github.com/samuelcolvin/pydantic/issues/1241
         # https://github.com/samuelcolvin/pydantic/issues/2763
         keep_untouched = (cached_property,)
+
+        arbitrary_types_allowed = True
+
+    def __json__(self) -> dict[Any, Any]:
+        return self.dict()
```

Comment:
Also define `__json__` for layer and dataset models? Currently full datasets are being repeated throughout the config, including all of their assets.

Reply:
would this look like def __json__(self) -> Layer[...]...?

Reply:
Resolved this by specifying `__json__` for the `Config` model. Excludes the `layers` and `datasets` elements; only the tree matters.